frontend/src/concepts/pipelines/apiHooks/usePipelines.ts

```diff
@@ -43,4 +49,44 @@ export const useSafePipelines = (
   );
 };
 
+async function getAllPipelines(
+  opts: K8sAPIOptions,
+  params: PipelineParams | undefined,
+  listPipelines: ListPipelines,
+): Promise<PipelineKFv2[] | undefined> {
+  const result = await listPipelines(opts, params);
+  let allPipelines = result.pipelines;
+
+  if (result.next_page_token) {
+    const nextPipelines = await getAllPipelines(
+      opts,
+      { pageToken: result.next_page_token },
+      listPipelines,
+    );
+    allPipelines = allPipelines?.concat(nextPipelines ?? []);
```

Comment:
Following my suggestions above, this could be `allPipelines = allPipelines.concat(nextPipelines);`